justice.lua

```diff
@@ -0,0 +1,40 @@
+
+local argparse = require('argparse')
+
+local function pardon_unit(unit)
+    for _,punishment in ipairs(df.global.plotinfo.punishments) do
+        if punishment.criminal == unit.id then
+            punishment.prison_counter = 0
+            return
+        end
+    end
+    qerror('Unit is not currently serving a sentence!')
+end
+
+local function command_pardon(unit_id)
+    local unit = nil
+    if not unit_id then
+        unit = dfhack.gui.getSelectedUnit()
+        if not unit then qerror("No unit selected!") end
+    else
+        unit = df.unit.find(unit_id)
+        if not unit then qerror(("No unit with id %i"):format(unit_id)) end
+    end
+    if unit then pardon_unit(unit) end
```

Comment:
```suggestion
    pardon_unit(unit)
```

Reply:
we still need this check because `unit` can still be `nil` if a unit wasn't found by either of the methods

Reply:
if a unit was not found in either of the if branches above, then `qerror` will throw